test/integration/framework/utils.go

```diff
@@ -0,0 +1,358 @@
+		if err != nil {
+			return nil, err
+		}
+	}
+	deployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: opts.Namespace,
+			Name:      o.TillerConfigurationOptions.DeploymentName,
+			Labels:    labels,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: opts.getReplicas(),
+			Selector: &metav1.LabelSelector{
+				MatchLabels: labels,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: labels,
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:            "tiller",
+							Image:           defaultTillerImage + ":canary",
+							ImagePullPolicy: opts.pullPolicy(),
+							Ports: []corev1.ContainerPort{
+								{ContainerPort: 44134, Name: "tiller"},
+								{ContainerPort: 44135, Name: "http"},
+							},
+							Env: []corev1.EnvVar{
+								{Name: "TILLER_NAMESPACE", Value: opts.Namespace},
+							},
+							LivenessProbe: &corev1.Probe{
+								Handler: corev1.Handler{
+									HTTPGet: &corev1.HTTPGetAction{
+										Path: "/liveness",
+										Port: intstr.FromInt(44135),
+									},
+								},
+								InitialDelaySeconds: 1,
+								TimeoutSeconds:      1,
+							},
+							ReadinessProbe: &corev1.Probe{
+								Handler: corev1.Handler{
+									HTTPGet: &corev1.HTTPGetAction{
+										Path: "/readiness",
+										Port: intstr.FromInt(44135),
+									},
+								},
+								InitialDelaySeconds: 1,
+								TimeoutSeconds:      1,
+							},
+						},
+					},
+					HostNetwork:  opts.EnableHostNetwork,
+					NodeSelector: nodeSelectors,
+				},
+			},
+		},
+	}
+	return deployment, nil
+}
+
+//TODO: add support to pod-prefixes filtering for logs
+func (o *GardenerTestOperation) fetchPodContainerLogs(podNamespace, podName, containerName string) (*string, error) {
+	logChannel := make(chan string, 1)
+
+	go func() {
+		var (
+			buffer *bytes.Buffer
+			err    error
+		)
+		if len(containerName) == 0 {
+			buffer, err = o.K8sShootClient.GetPodLogs(podNamespace, podName, &corev1.PodLogOptions{})
+		} else {
+			buffer, err = o.K8sShootClient.GetPodLogs(podNamespace, podName, &corev1.PodLogOptions{
+				Container: containerName,
+			})
+		}
+		if err != nil {
+			o.ShootGardenerTest.Logger.Errorf("Something went wrong fetching logs: %s", err.Error())
+			return
+		}
+		logToReturn := buffer.String()
+		logChannel <- logToReturn
+	}()
+
+	select {
+	case result := <-logChannel:
+		return &result, nil
+	case <-time.After(2 * time.Minute):
+		return nil, ErrTimeoutWaitingWaitingForLogs
+	}
+}
+
+// getRunningPodByLabels fetches the first running pod with the desired set of labels <labelsMap>
+func (o *GardenerTestOperation) getRunningPodByLabels(labelsMap map[string]string, seedOrShoot string) (*corev1.Pod, error) {
+	var runningPodWithDesiredLabels *corev1.Pod
+	set := labels.Set(labelsMap)
+
+	var podList *corev1.PodList
+	var err error
+
+	switch seedOrShoot {
+	case "seed":
```

Comment:
Don't switch on a `seedOrShoot` string, make a 'go enum' for this (type + constants) and determine the namespace depending on this. Then you don't have that much duplicate code.